src/widgetastic/widget/table.py

```diff
@@ -491,12 +491,12 @@ def clear_cache(self):
             "header_index_mapping",
             "index_header_mapping",
             "assoc_column_position",
-            "table_tree",
```

Comment:
Can you provide more context about the error you might be encountering, and what else you might have considered at this point?  With some of this data maybe I can help you write a unit test to replicate the failure and validate the fix!

All of the other labels here are also instance properties, not class or instance attributes.  So I don't think it being a property is the sole issue.

The difference between the rest of the labels and `Table.table_tree` is that `table_tree` method isn't decorated as a cached property.  

I'm guessing what you've seen is the manifestation of this behavior, where `cached_property` values are added to the instance's `__dict__`, and are thus resolved as attributes. The non-cached properties are not attributes even after being called.


```
In [6]: class TestProp():
   ...:     def __init__(self, arg):
   ...:         self._value = arg
   ...:     @property
   ...:     def regular(self):
   ...:         return self._value
   ...:     @cached_property
   ...:     def cached(self):
   ...:         self._value = '0'
   ...:         return self._value
   ...: 

In [7]: t = TestProp(1)

In [8]: t._value
Out[8]: 1

In [9]: t.__dict__
Out[9]: {'_value': 1}

In [10]: t.regular
Out[10]: 1

In [11]: t.__dict__
Out[11]: {'_value': 1}

In [12]: t.cached
Out[12]: '0'

In [13]: t.__dict__
Out[13]: {'_value': '0', 'cached': '0'}
```


In principle this change does appear to be correct, since `Table.table_tree` isn't an attribute like the rest of the labels in this list, it can't be deleted, and deleting the instance attribute that controls the property method behavior would effectively clear the cache.


I'd like to hear thoughts on whether `Table.table_tree` should be a cached property like the rest of these.

Reply:
> Can you provide more context about the error you might be encountering, and what else you might have considered at this point?
I was dealing with a paginated table. In the first page it was having 10 rows and I can loop through them. However, when I get to the last page, which was having less rows, and I try to loop through them, it also tries to loop through 10 rows (regardless of how many there are). 
I though the `clear_cache` method was the solution to this problem, but it wasn't. When calling it I found that the `table_tree` always raises the `AttributeError` exception. If it always raises the exception, then I do not see the point of having it defined in the list of items to be deleted. I think it should be deleted in a different way.
I have proposed to do `self._table_tree = None` because when setting the [table_tree method](https://github.com/RedHatQE/widgetastic.core/blob/d458add2e8fe7b2141e00a6dc9f6c0ea92707d62/src/widgetastic/widget/table.py#L444) to return `None`, my issue was gone; and it is a change that does not modify the definition of the exposed objects.
Said that, if you think setting `Table.table_tree` as a `cache_property` is a better solution, I will be happy to add that change (I was just a bit reluctant to modify exposed objects for a property I am not pretty sure how it is used).

Thank you for the explanation of `property` vs `cached_property` within objects `__dict__`.